src/plugins/cosmos/components/StakingInput/StakingInput.tsx

```diff
@@ -0,0 +1,128 @@
+import { Flex, FlexProps } from '@chakra-ui/layout'
+import { Input, InputLeftElement, Tag } from '@chakra-ui/react'
+import { Asset } from '@shapeshiftoss/types'
+import { Field, StakingValues } from 'plugins/cosmos/components/modals/Staking/views/Stake'
+import { Control, Controller } from 'react-hook-form'
+import NumberFormat from 'react-number-format'
+import { useLocaleFormatter } from 'hooks/useLocaleFormatter/useLocaleFormatter'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+
+type StakingInputProps = {
+  isCryptoField: boolean
+  onInputToggle: () => void
+  onInputChange: (value: string) => void
+  amountRef: string | null
+  asset: Asset
+  control: Control<StakingValues>
+}
+
+const cryptoInputValidation = {
+  required: true,
+  validate: {
+    validateCryptoAmount: (_: string) => {
+      // TODO: Implement when we have cosmos/osmosis balance data
+      return true
+    }
+  }
+}
+const fiatInputValidation = {
+  required: true,
+  validate: {
+    validateFiatAmount: (_: string) => {
+      // TODO: Implement when we have cosmos/osmosis balance data
+      return true
+    }
+  }
+}
+const CryptoInput = (props: any) => (
+  <Input
+    height='100%'
+    pr='4.5rem'
+    pl='1rem'
+    ml='1rem'
+    size='lg'
+    type='number'
+    border={0}
+    borderBottomRadius={0}
+    borderTopLeftRadius={0}
+    placeholder='Enter amount'
+    {...props}
+  />
+)
+
+export const StakingInput = ({
+  asset,
+  amountRef,
+  control,
+  isCryptoField,
+  onInputToggle,
+  onInputChange,
+  ...styleProps
+}: StakingInputProps & FlexProps) => {
+  const {
+    number: { localeParts }
+  } = useLocaleFormatter({ fiatType: 'USD' })
+  return (
```

Comment:
we can do `if (!amountRef) return null` to avoid casting below

Reply:
Not possible currently since the access is done two-callbacks deep and the narrowing gets lost by then 😢 .
Checked by trying to access `amountref` directly after this check and it gives the right narrowed type.